Library/Homebrew/brew.sh

```diff
@@ -195,19 +195,30 @@ then
   HOMEBREW_BASH_COMMAND="$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh"
 fi
 
-if [[ "$(id -u)" = "0" && "$(/usr/bin/stat -f%u "$HOMEBREW_BREW_FILE")" != "0" ]]
-then
+check-run-command-as-root() {
   case "$HOMEBREW_COMMAND" in
-    analytics|install|reinstall|postinstall|link|pin|update|upgrade|vendor-install|create|migrate|tap|tap-pin|switch)
+    analytics|install|reinstall|postinstall|link|pin|update|upgrade|\
+    vendor-install|create|migrate|tap|tap-pin|switch)
+      ;;
+    *)
+      return
+      ;;
+  esac
+
+  [[ "$(id -u)" = 0 ]] || return
+
+  local brew_file_ls_info=($(ls -nd "$HOMEBREW_BREW_FILE"))
```

Comment:
Thanks! I can see an argument for using `ls`. Hopefully few people are running `brew` as root (I never do) and even fewer with a `brew` command we don't want them to run as root, so this check will need to be performed fairly infrequently.

I'm glad we were able to rearrange this, so this isn't shelling out every single time to `id`, even if the command being invoked doesn't care at all whether it is invoked as root or not.